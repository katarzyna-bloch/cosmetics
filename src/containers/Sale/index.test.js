import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Sale } from '.'

jest.mock('../../components/ProductsList', () => {
  const ProductsList = () => null
  return ProductsList
})

it('should render correctly', () => {
  const props = {
    fetchProducts: jest.fn(),
    products: [
      { 
      name: 'Test',
      img: 'img.jpg',
      id: '1',
      price: '10zł',
      oldPrice: '49zł',
      hasDiscount: true,
      },
    ],
  }
  const wrapped = mount(<Sale {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  wrapped.update()
  expect(props.fetchProducts).toHaveBeenCalled()
})
