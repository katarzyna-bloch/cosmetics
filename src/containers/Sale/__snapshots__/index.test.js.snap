// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Sale
  fetchProducts={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  products={
    Array [
      Object {
        "hasDiscount": true,
        "id": "1",
        "img": "img.jpg",
        "name": "Test",
        "oldPrice": "49zł",
        "price": "10zł",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdvvaa hugagk"
    >
      <styled.h5>
        <h5
          className="sc-gsDJrp ilHCrl"
        >
          Promocje, które zachwycą każdego
        </h5>
      </styled.h5>
      <ProductsList
        products={
          Array [
            Object {
              "hasDiscount": true,
              "id": "1",
              "img": "img.jpg",
              "name": "Test",
              "oldPrice": "49zł",
              "price": "10zł",
            },
          ]
        }
      />
    </div>
  </styled.div>
</Sale>
`;
